client/asset/interface.go

```diff
@@ -473,14 +476,38 @@ type Broadcaster interface {
 type Bonder interface {
 	Broadcaster
 
+	// RegisterUnspent informs the wallet of a certain amount already locked in
+	// unspent bonds that will eventually be refunded with RefundBond. This
+	// should be used prior to ReserveBondFunds. This alone does not enable
+	// reserves enforcement, and it should be called on bring-up when existing
+	// bonds that may refund to this wallet are known. Once ReserveBondFunds is
+	// called, these live bond amounts will become enforced reserves when they
+	// are refunded via RefundBond.
+	RegisterUnspent(live uint64)
+	// ReserveBondFunds (un)reserves funds for creation of future bonds.
+	// MakeBondTx will create transactions that decrease these reserves, while
+	// RefundBond will replenish the reserves. If the wallet's available balance
+	// should be respected when adding reserves, the boolean argument may be set
+	// to indicate this, in which case the return value indicates if it was able
+	// to reserve the funds. In this manner, funds may be pre-reserved so that
+	// when the wallet receives funds (from either external deposits or
+	// refunding of live bonds), they will go directly into locked balance. When
+	// the reserves are decremented to zero (by the amount that they were
+	// incremented), all enforcement including any fee buffering is disabled.
+	ReserveBondFunds(future int64, respectBalance bool) bool
+
```

Comment:
We have that unused `asset.Balance.Other` field.

I expect frontend can work with the following or similar?

```diff
@@ -1061,7 +1062,9 @@ func (dcr *ExchangeWallet) Balance() (*asset.Balance, error) {
                dcr.log.Warnf("Available balance is below configured reserves: %f < %f",
                        toDCR(bal.Available), toDCR(reserves))
                reserves = bal.Available
+               bal.Other["Reserves Deficit"] = reserves - bal.Available
        }
+       bal.Other["Reserves"] = reserves
        bal.Available -= reserves
        bal.Locked += reserves
```

Or keys can be prefixed like `bal.Other["warning:Reserves Deficit"] = reserves - bal.Available` so it can be shown in red or something?

Reply:
I this could work out

![image](https://user-images.githubusercontent.com/9373513/220485924-8bde0950-2022-4b0b-ab19-10b3c68e3d38.png)

I'd update that with `Bonds (locked)` too for completeness

Reply:
I kinda like this, in the last commit

![image](https://user-images.githubusercontent.com/9373513/220500955-be813c8b-054e-48ce-b0f0-12cd270f4108.png)

Reply:
Maybe `Bonded` and `Bond Reserves`. Someone might not know what the reserves are for.